### ISSUE TITLE
mb_detect_encoding() expects parameter 1 to be string, null given

### ISSUE THREAD

**Issue by jupitern** [Respect/Validation#1328]
Hi,
fter updating to v2 and fixing some issues I still have this one that I don't understand. Nowhere on my code I pass a null value to this validator. Can anyone help? thanks.

Error:
````
mb_detect_encoding() expects parameter 1 to be string, null given on /var/www/html/vendor/respect/validation/library/Rules/StartsWith.php line 71~
````

Stacktrace:
````
#0 /var/www/html/vendor/respect/validation/library/Rules/StartsWith.php(71): mb_detect_encoding()
#1 /var/www/html/vendor/respect/validation/library/Rules/StartsWith.php(59): Respect\Validation\Rules\StartsWith->validateEquals()
#2 /var/www/html/vendor/respect/validation/library/Rules/AbstractRule.php(43): Respect\Validation\Rules\StartsWith->validate()
#3 /var/www/html/vendor/respect/validation/library/Rules/AbstractComposite.php(102): Respect\Validation\Rules\AbstractRule->assert()
#4 [internal function]: Respect\Validation\Rules\AbstractComposite->Respect\Validation\Rules\{closure}()
#5 /var/www/html/vendor/respect/validation/library/Rules/AbstractComposite.php(111): array_map()
#6 /var/www/html/vendor/respect/validation/library/Rules/AllOf.php(31): Respect\Validation\Rules\AbstractComposite->getAllThrownExceptions()
#7 /var/www/html/vendor/respect/validation/library/Rules/AbstractComposite.php(102): Respect\Validation\Rules\AllOf->assert()
#8 [internal function]: Respect\Validation\Rules\AbstractComposite->Respect\Validation\Rules\{closure}()
#9 /var/www/html/vendor/respect/validation/library/Rules/AbstractComposite.php(111): array_map()
````

**Comment by servdebtdev** [comment]
This is the code that results in a exception. If I am using oneOf validation and the first fails it goes and test tests the startsWith validation that should start by validation if type is string

````
(new v())
            ->key("chatid", v::oneOf(v::nullType(), v::startsWith("chat.")))
            ->assert(["chatid" => null]);
````

**Comment by henriquemoody** [comment]
This is fixed on version 2.2.1, please upgrade your version. 🐼
